src/Encoding/FilteredStream.php

```diff
@@ -174,15 +179,17 @@ public function isSeekable()
      */
     public function rewind()
     {
-        throw new \RuntimeException('Cannot rewind a filtered stream');
+        @trigger_error('Filtered streams are not seekable. This method will start raising an exception in the next major version');
```

Comment:
Maybe pass `E_USER_DEPRECATED` as the second argument to handle it as a normal deprecation?

Reply:
IMHO it's not a deprecation, but rather being nice to people using a buggy behavior

Reply:
I agree with Joel's reasoning, but raising it as a deprecation might help when using tools that detect deprecations.

Reply:
changed from default (E_USER_NOTICE) to E_USER_DEPRECATED for better visibility